pontoon/base/views.py

```diff
@@ -76,34 +81,81 @@ def home(request):
     return translate(request, locale, project.slug)
 
 
-def locale(request, locale, template='locale.html'):
-    """Locale view."""
-    log.debug("Locale view.")
+class TemplateModelView(TemplateView):
+    def get(self, request, *args, **kwargs):
+        try:
+            return super(TemplateModelView, self).get(request, *args, **kwargs)
+        except ObjectDoesNotExist:
+            return self.object_does_not_exist(request, *args, **kwargs)
 
-    # Validate locale
-    try:
-        l = Locale.objects.get(code__iexact=locale)
-    except Locale.DoesNotExist:
-        raise Http404
+    def object_does_not_exist(self, *args, **kwargs):
+        return Http404()
 
-    projects = Project.objects.filter(
-        disabled=False, pk__in=Resource.objects.values('project'), locales=l) \
-        .order_by("name")
 
-    if not projects:
-        messages.error(
-            request, "Oops, no projects available for this locale.")
-        request.session['translate_error'] = {
+class ContributorsView(TemplateView):
+    """
+    View returns top contributors.
+    """
+    template_name = 'users.html'
+
+    def get_context_data(self, **kwargs):
+        """Top contributors view."""
+        log.debug("Top contributors view.")
+
+        context = super(ContributorsView, self).get_context_data(**kwargs)
+        try:
+            period = int(self.request.GET['period'])
+            if period <= 0:
+                raise ValueError
+            start_date = (timezone.now() + relativedelta(months=-period))
+        except (KeyError, ValueError):
+            period = None
+            start_date = None
+
+        context['contributors'] = User.translators.with_translation_counts(start_date, kwargs.get('query_args'))
```

Comment:
The preferred way of passing something like the `query_args` down to subclasses is to make an attribute or method on the class that parent classes can overload. So instead of having classes that inherit from `ContributorsView` override `get_context_data` and pass in the extra filters they want to apply, you'd have either a `translation_query_filters` attribute or `get_translation_query_filters` method for them to override. Then just call/reference that here and they can avoid overriding the unrelated `get_context_data` method.